tools/testdxildll/DxilContainerBuilder.cpp

```diff
@@ -0,0 +1,33 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// dxcontainerbuilder.cpp                                                    //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Implements the Dxil Container Builder                                     //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxc/DxilContainer/DxcContainerBuilder.h"
+#include "dxc/DxilContainer/DxilContainer.h"
+#include "dxc/Support/ErrorCodes.h"
+#include "dxc/Support/FileIOHelper.h"
+#include "dxc/Support/Global.h"
+#include "dxc/Support/WinIncludes.h"
+#include "dxc/Support/dxcapi.impl.h"
+#include "dxc/dxcapi.h"
+
+#include <algorithm>
+
+using namespace hlsl;
+
+HRESULT CreateDxcContainerBuilder(_In_ REFIID riid, _Out_ LPVOID *ppv) {
```

Comment:
Couldn't this be implemented in DxilLib.cpp, since we really aren't supplying another implementation here?  One might almost mix it up with `DxcContainerBuilder.cpp` where we implement the internal builder that this code uses.

Reply:
There were some compilation errors when I tried to do this before that had to do with conflicting defines. I'll take another run at it.